common/courses/participants.ts

```diff
@@ -146,23 +148,41 @@ export async function canJoinSubcourse(subcourse: Subcourse, pupil: Pupil): Prom
     return { allowed: true };
 }
 
-// Whether the pupil could join the course if the course was not full (i.e. pupils can still join the waitinglist)
-export async function couldJoinSubcourse(subcourse: Subcourse, pupil: Pupil): Promise<Decision<CourseDecision>> {
-    if (!canJoinSubcourses(pupil).allowed) {
-        return canJoinSubcourses(pupil);
-    }
-
+// Whether a subcourse is in principle joinable at this point in time, not considering the number of participants
```

Comment:
What's the difference to couldJoinSubcourse? According to comments, both seem to check if a pupil can join the subcourse regardless of whether there are still free course places.

Reply:
Yeah we always have a permutation of some of the checks and some are skipped. Could potentially be refactored at some point.